plugins/baseposevelocity/src/BasePoseVelocityDriver.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include "BasePoseVelocityDriver.h"
+
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+
+
+using namespace yarp::dev;
+
+GazeboYarpBasePoseVelocityDriver::GazeboYarpBasePoseVelocityDriver() : m_robot(0), m_baseLink(0)
+{
+
+}
+
+GazeboYarpBasePoseVelocityDriver::~GazeboYarpBasePoseVelocityDriver()
+{
+    if (m_robot)
+    {
+        m_robot = nullptr;
+    }
+    
+    if (m_baseLink)
+    {
+        m_baseLink = nullptr;
+    }
+}
+
+bool GazeboYarpBasePoseVelocityDriver::open(yarp::os::Searchable& config)
+{
+    yarp::os::Property deviceProp;
+    deviceProp.fromString(config.toString().c_str());
+       
+    std::string robot(deviceProp.find("robot").asString().c_str());
+    m_robot = GazeboYarpPlugins::Handler::getHandler()->getRobot(robot);
+    if (m_robot == NULL)
+    {
+        yError() << "GazeboYarpBasePoseVelocityDriver: robot model not found";
+        return false;
+    }
+
+    m_baseLinkName = deviceProp.find("baseLink").asString().c_str();
+    std::string linkNameScopedEnding = "::" + m_baseLinkName;
+    const gazebo::physics::Link_V &links = m_robot->GetLinks();
+    for (size_t idx = 0; idx < links.size(); idx++)
+    {
+      std::string linkName = links[idx]->GetScopedName();
+      if (GazeboYarpPlugins::hasEnding(linkName, linkNameScopedEnding))
+      {
+        m_baseLink = m_robot->GetLink(linkName);
+        break;
+      }
+    }
+    
+    if (m_baseLink == nullptr)
+    {
+        yError() << "GazeboYarpBasePoseVelocityDriver: link not found in the model";
+        return false;
+    }
+    
+    m_baseState.resize(m_stateDimensions);
+    
+    m_updateConnection = gazebo::event::Events::ConnectWorldUpdateBegin(boost::bind(&GazeboYarpBasePoseVelocityDriver::onUpdate, this, _1));
+    
+    return true;
+}
+
+bool GazeboYarpBasePoseVelocityDriver::close()
+{
+    yarp::os::LockGuard guard(m_dataMutex);
+    this->m_updateConnection.reset();
+    return true;
+}
+
+void GazeboYarpBasePoseVelocityDriver::onUpdate(const gazebo::common::UpdateInfo& _info)
+{
+    ignition::math::Vector3d m_worldBasePosition = m_baseLink->WorldPose().Pos();
+    ignition::math::Quaterniond m_worldBaseOrientation = m_baseLink->WorldPose().Rot();
+    ignition::math::Vector3d m_worldBaseLinVel = m_baseLink->WorldLinearVel();
+    ignition::math::Vector3d m_worldBaseAngVel = m_baseLink->WorldAngularVel();
+    ignition::math::Vector3d m_worldBaseLinAcc = m_baseLink->WorldLinearAccel();
```

Comment:
Not super relevant, but please be aware of https://bitbucket.org/osrf/gazebo/issues/893/imusensor-should-get-acceleration-directly#comment-19435791 .

Reply:
The first step should be open a proper issue in Gazebo, but I think it is ok not do anything, I just want you to be aware of this. Once we have a proper Gazebo issue, we can link it in the documentation.